pkg/observer/controller.go

```diff
@@ -243,6 +247,63 @@ func (c *controller) dropInProgressNodeGroups(nodeGroups v1.NodeGroupList, cnrs
 	return restingNodeGroups
 }
 
+// get the cluster-autoscaler last scaleUp activity time
+func stringToTime(s string) (time.Time, error) {
+	sec, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return time.Unix(sec, 0), nil
+}
+
+// query cluster-autoscaler metrics to figure out if it's safe to start a new CNR
+func (c *controller) safeToStartCycle() bool {
+	safeToStart := true
+	client, err := api.NewClient(api.Config{
+		Address: c.PrometheusAddress,
+	})
+	if err != nil {
+		// Prometheus might not be installed in the cluster. return true if it can't connect
+		klog.Errorln("Error creating client:", err)
+		return safeToStart
+	}
+
+	v1api := promv1.NewAPI(client)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	result, warnings, err := v1api.Query(ctx, "cluster_autoscaler_last_activity{activity='scaleUp'}", time.Now())
+	if err != nil {
+		// cluster-autoscaler might not be installed in the cluster. return true if it can't find the metrics of run the query
+		klog.Errorln("Error querying Prometheus: ", err)
+		return safeToStart
+	}
+	if len(warnings) > 0 {
+		klog.Errorln("Warnings: ", warnings)
+	}
+
+	v := result.(model.Vector)
+	// consider safe if there isn't any scaleUp event yet
+	if v.Len() == 0 {
+		klog.Infoln("No scaleUp events")
+		return safeToStart
+	}
+
+	scaleUpTime := v[v.Len()-1].Value.String()
+	t, err := stringToTime(scaleUpTime)
+	if err != nil {
+		klog.Fatalln("Error converting the time: ", err)
+	}
+
+	lastScaleEvent := time.Since(t)
+	if lastScaleEvent <= c.NodeStartupTime {
+		safeToStart = false
+		klog.Infoln("Scale up event recently happened", c.NodeStartupTime)
+		return safeToStart
```

Comment:
no need to keep the state in a variable here 

`return false`

Reply:
👍